File: ios-objc/tooling/RIBs.xctemplate/Default/___FILEBASENAME___Builder.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@implementation ___VARIABLE_productName___Builder

- (instancetype)initWithDependency:(id<___VARIABLE_productName___Dependency>)dependency {
    return [super initWithDependency:dependency];
}

- (id<___VARIABLE_productName___Routing>)buildWithListener:(id<___VARIABLE_productName___Listener>)listener {
    ___VARIABLE_productName___Component *component = [[___VARIABLE_productName___Component alloc] initWithDependency:self.dependency];
    ___VARIABLE_productName___Interactor *interactor = [___VARIABLE_productName___Interactor new];
    interactor.listener = listener;
    // TODO: Make sure to remove the prefix and convert the variable into lower-camelcase.
    return [[___VARIABLE_productName___Router alloc] initWithInteractor:interactor viewController:component.___VARIABLE_productName___ViewController childBuilders:[self childWithDependency:component]];
}

+ (nullable NSArray<Class<BRKBuildable>> *)childBuilders {
    /// if your builder has no child RIB. please delete this method
    /// you can insert your child RIB builder class in this array
    return @[];
}

@end

File: ios/tooling/RIB.xctemplate/Default/___FILEBASENAME___Builder.swift
//___FILEHEADER___

import RIBs

protocol ___VARIABLE_productName___Dependency: Dependency {
    // TODO: Make sure to convert the variable into lower-camelcase.
    var ___VARIABLE_productName___ViewController: ___VARIABLE_productName___ViewControllable { get }
    // TODO: Declare the set of dependencies required by this RIB, but won't be
    // created by this RIB.
}

final class ___VARIABLE_productName___Component: Component<___VARIABLE_productName___Dependency> {

    // TODO: Make sure to convert the variable into lower-camelcase.
    fileprivate var ___VARIABLE_productName___ViewController: ___VARIABLE_productName___ViewControllable {
        return dependency.___VARIABLE_productName___ViewController
    }

    // TODO: Declare 'fileprivate' dependencies that are only used by this RIB.
}

// MARK: - Builder

protocol ___VARIABLE_productName___Buildable: Buildable {
    func build(withListener listener: ___VARIABLE_productName___Listener) -> ___VARIABLE_productName___Routing
}

final class ___VARIABLE_productName___Builder: Builder<___VARIABLE_productName___Dependency>, ___VARIABLE_productName___Buildable {

    override init(dependency: ___VARIABLE_productName___Dependency) {
        super.init(dependency: dependency)
    }

    func build(withListener listener: ___VARIABLE_productName___Listener) -> ___VARIABLE_productName___Routing {
        let component = ___VARIABLE_productName___Component(dependency: dependency)
        let interactor = ___VARIABLE_productName___Interactor()
        interactor.listener = listener
        return ___VARIABLE_productName___Router(interactor: interactor, viewController: component.___VARIABLE_productName___ViewController)
    }
}
